### DISCUSSION TITLE
Unable to call ListUser function in zitadel from zitadel-go SDK

### DISCUSSION THREAD

**Discussion by prajwalpamin** [zitadel/zitadel#9095] [Q&A]
Hi Team , I'm using a saas server written in go to make call to ListUser function here is the implementation of my code for the same (zitadel-go sdk version 3.3.1 and zitadel version 2.66.0). 
Note: Im running saas server and zitadel both in on-prem environment and APIENDPOINT and ISSUERENDPOINT are setup.

SAAS api below
**func UpdateUserStatus(ctx context.Context, in *pb2.UpdateUserStatusRequest) *pb2.UpdateUserStatusResponse {
	if in.OrgId == "" {
		log.Print("Error occurred OrgId cannot be empty")
		return &pb2.UpdateUserStatusResponse{
			Response: "OrgId missing in payload",
			Success:  false,
		}
	}
	tlsConfig := &tls.Config{
		InsecureSkipVerify: true,
	}
	transportOption := grpc.WithTransportCredentials(credentials.NewTLS(tlsConfig))

	client, err := userv2.NewClient(ctx,
		constants.ISSUERENDPOINT,
		constants.APIENDPOINT,
		[]string{oidc.ScopeOpenID, zitadel.ScopeZitadelAPI()},
		zitadel.WithJWTProfileTokenSource(middleware.JWTProfileFromKeyAndUserID(ctx,
			common.S2b(apiSecret.Key),
			apiSecret.KeyID,
			apiSecret.UserID,
		)),
		zitadel.WithOrgID(in.OrgId),
		zitadel.WithDialOptions(transportOption),
	)

	if err != nil {
		log.Printf("Error occurred connecting to zitadel instance %v", err.Error())
		return &pb2.UpdateUserStatusResponse{
			Response: "INTERNAL SERVER ERROR",
			Success:  false,
		}
	}

	defer client.Connection.Close()

	userId := ""
	split := strings.Split(in.GetId(), ":")
	email := in.GetId()
	empCode := ""
	if len(split) > 1 {
		email = split[0]
		empCode = split[1]
	}
	log.Println(email, empCode)
	temp := []*user.SearchQuery{}
	temp = append(temp, &user.SearchQuery{
		Query: &user.SearchQuery_UserNameQuery{
			UserNameQuery: &user.UserNameQuery{
				UserName: empCode,
				Method:   1,
			},
		},
	})
	log.Printf("Searching Id for User with Email: %v split values %v", in.GetId(), split)
	response, err := client.ListUsers(ctx, &user.ListUsersRequest{
		Queries: temp,
	})

	if err != nil {
		log.Printf("Error occurred getting user by empCode %v", err.Error())
	}
	if response == nil || response.Result == nil || len(response.Result) == 0 {
		temp1 := []*user.SearchQuery{}
		temp1 = append(temp1, &user.SearchQuery{
			Query: &user.SearchQuery_UserNameQuery{
				UserNameQuery: &user.UserNameQuery{
					UserName: email,
					Method:   1,
				},
			},
		})
		response, err = client.ListUsers(ctx, &user.ListUsersRequest{
			Queries: temp1,
		})
		if err != nil {
			log.Printf("Error occurred getting user by email %v", err.Error())
			return &pb2.UpdateUserStatusResponse{
				Response: fmt.Sprintf("Error occurred getting user by email %v", err.Error()),
				Success:  false,
			}
		}
	}
	if response.Result != nil && len(response.Result) > 0 {
		if response.Result[0].State == user.UserState_USER_STATE_INITIAL {
			return &pb2.UpdateUserStatusResponse{
				Response: "USER_YET_TO_INITIALIZE",
				Success:  false,
			}
		}
		userId = response.Result[0].UserId
	}

	errStr := ""
	// Force true Activate
	if in.Force {
		log.Printf("Request to Activate user email %v and Id %v", in.GetId(), userId)
		_, err = client.ReactivateUser(ctx, &user.ReactivateUserRequest{
			UserId: userId,
		})
		if err != nil {
			errStr = err.Error()
			log.Printf("Error occurred removing user by email2 %v", errStr)
			if strings.Contains(errStr, "User is not inactive") {
				return &pb2.UpdateUserStatusResponse{
					Response: "Successful",
					Success:  true,
				}
			}
			return &pb2.UpdateUserStatusResponse{
				Response: errStr,
				Success:  false,
			}
		}
	} else {
		log.Printf("Request to Deactivate user email %v and Id %v", in.GetId(), userId)
		_, err = client.DeactivateUser(ctx, &user.DeactivateUserRequest{
			UserId: userId,
		})
		if err != nil {
			errStr = err.Error()
			log.Printf("Error occurred removing user by email1 %v", err.Error())
			if strings.Contains(errStr, "User already inactive") {
				return &pb2.UpdateUserStatusResponse{
					Response: "Successful",
					Success:  true,
				}
			}
			return &pb2.UpdateUserStatusResponse{
				Response: errStr,
				Success:  false,
			}
		}
	}

	return &pb2.UpdateUserStatusResponse{
		Response: "Successful",
		Success:  true,
	}
}**

The error message in my saas api is 
 **Error occurred getting user by empCode rpc error: code = Unimplemented desc = unexpected HTTP status code received from server: 404 (Not Found); transport: received unexpected content-type "application/json"**

and in zitadel logs i get 
**Failed to extract server metadata from context**
Kindly help me to resolve this issue

**Comment by fforootd** [comment]
Hm that 404 is weird. I think it could have something to do with a http2 problem in the proxy.

Can you share how you deployed Zitadel and if there is a reverse proxy in front of Zitadel?

There is some guide for proxies [here](https://zitadel.com/docs/self-hosting/manage/reverseproxy/reverse_proxy).
Please make sure that http2 or h2c is activated in the upstream.
